server/utils/conditionsProvider.ts

```diff
@@ -35,3 +37,156 @@ export function getGroupedAdditionalConditions(): Record<string, unknown>[] {
   })
   return Array.from(map, ([category, conditions]) => ({ category, conditions }))
 }
+
+/**
+ * Accepts a list of additional conditions and data items for a licence and expands the
+ * placeholders with their matching data to produce a list of formatted additional conditions.
+ * @param conditions
+ */
+export function expandAdditionalConditions(conditions: AdditionalCondition[]): string[] {
+  const expandedConditions: string[] = []
+
+  conditions?.forEach(condition => {
+    const configCondition = getAdditionalConditionByCode(condition.code)
+    if (configCondition?.requiresInput) {
+      const placeholders = getPlaceholderNames(configCondition.tpl as string)
+      const inputConfig = configCondition.inputs as unknown as Record<string, unknown>[]
+      let thisConditionText = configCondition.tpl as string
+
+      for (let i = 0; i < placeholders.length; i += 1) {
```

Comment:
`placeholders.forEach()` ?

Reply:
I did this first time around, however it never goes into the loop! Something odd going on with Array.forEach in these circumstances, which I was thinking I would come back to once the code and tests were in and working.

Reply:
Any ideas appreciated why the .forEach might not work here?

Reply:
Done now - next PR.